editor/components/block-list/block.js

```diff
@@ -603,7 +604,7 @@ export class BlockListBlock extends Component {
 					rootUID={ rootUID }
 					layout={ layout }
 				/>
-				{ shouldShowMovers && (
+				{ shouldShowMovers && ! showSideInserter && (
```

Comment:
Seems like `! showSideInserter` ought to be built-in to the condition assigning to `shouldShowMovers`, if we're expecting them to be mutually exclusive (prevents future errors where this is not accounted for).